### PR TITLE
Feature/unit test parse revid

### PR DESCRIPTION
**Pull request by tleyden**

### CODE REVIEW

**Comment by tleyden** [comment]
This adds a unit test for ParseRevID.  Currently passing. 

Cherry picked from work in progress: https://github.com/couchbase/sync_gateway/issues/2651

File: db/revision_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"testing"
+	"log"
+)
+
+func TestParseRevID(t *testing.T) {
+
+	var generation int
+	var digest string
+
```
Comment (adamcfraser) [inline db/revision_test.go:12]: A revision's digest is an MD5 checksum in hex (as string), so should only have the characters `[a-fA-F0-9]`.

Our current ParseRevID doesn't enforce that, but it's probably reasonable to limit the tests to those values (i.e. an alternate implementation of ParseRevID doesn't actually need to handle something like 2222-lj-lkjl).

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/revision_test.go (hunk 1 above)
Comment (tleyden) [reply db/revision_test.go:12]: OK I went ahead and removed that part of the test

**Comment by tleyden** [comment]
@adamcfraser this one should be ready to merge

**Pull request merged**